### PR TITLE
EOM-CC

### PR DESCRIPTION
**Pull request by ajay-mk**
## Implements derivation of EOM-CC methods. 
Currently only supports EE-EOM-CC. IP and EA are disabled for now.

### Checklist
- [x] Add operator level support for L and R operators in ~`mbpt::sr::op`~ `mbpt` namespace
- [x] Refactor `P()` and `A()` operators to support non-particle conserving projectors
- [x] Implement derivation of left and right EOM equations using `CC::eom_l()` and `CC::eom_r()`
- [x] Extend unit tests
- [x] Cleanup in `op.hpp` and `op.cpp`
- [x] Update vocabulary in `OpMaker` and tensor and op level functions
#### Misc changes
- [x] Factor out `lower_to_tensor_form` to be a standalone function
- [x] Update action versions in CI to hush warnings
- [x] Documentation update in several places

### CODE REVIEW

**Comment by ajay-mk** [comment]
@evaleev This is ready for review. I will merge #191 and #194 once they are merged into master.

**Comment by evaleev** [comment]
I think it will be easier if we merge #187 first (since it is a foundational refactor) then update this in response.

**Review by evaleev — APPROVED** [review]

**Pull request merged**
